### DISCUSSION TITLE
How change route name from nuxt-config.

### DISCUSSION THREAD

**Discussion by andysay** [nuxt/nuxt#17775] [Questions] [answered]
in pages dir i have file test.vue 

in nuxt.config      
```
   'pages:extend'(pages) {
            pages.push({
                name: 'changename,
                path: '/test',
                file: resolve(__dirname, 'pages/test.vue'),
            });
        },
```
but name not change ( how i can make it ?

**Comment by danielroe** [comment, marked as answer]
You should find the existing route in the `pages` array and change it there.

**Reply by guzz** [reply]
I'm having some instability with this solution... My use case is a little different as I am trying to build an i18n plugin, so my approach is to get all the pages registered and duplicate them with a new path and name, but still point to the same file. 

After this, the routes are created but when I try to navigate to the localized page it redirects me to "/:locale/:locale/{my-path}" and it breaks nitro with an error: [worker] Cannot set headers after they are sent to the client.

Is there another approach that would be more compatible with how nuxt handles routing?

Thanks in advance

**Reply by danielroe** [reply]
@guzz I'd recommend you share your code so we can see what the issue is.

**Reply by guzz** [reply]
Sure @danielroe, here are the 2 repositories for this: 
- the api: https://github.com/guzz/strapi-blog-example
- the nuxt-app: https://github.com/guzz/nuxt-blog-example

I just a simple blog for educational purposes using apollo GraphQL

**Comment by andysay** [comment]
ok i did it ) thanks!
